src/plugins/rocket.rs

```diff
@@ -1,34 +1,69 @@
-            .get("Authorization")
-            .collect::<Vec<&str>>()
-        {
-            auth_field if auth_field.is_empty() => outcome::Outcome::Failure((
-                Status::BadRequest,
-                Error::InvalidAuthHeader(InvalidAuthHeader::MissingAuthHeader),
-            )),
-            auth_field if auth_field.len() == 1 => {
-                auth_field[0].try_into().into_outcome(Status::BadRequest)
+        // Is header present?
+        match req.headers().get_one("Authorization") {
+            Some(header_raw) => {
+                let (header_name, header_content) = header_raw.split_at(7);
+                // Is header in the correct format?
+                if header_name != "Bearer " {
+                    request::Outcome::Failure((
+                        Status::Unauthorized,
+                        Error::InvalidAuthHeader(InvalidAuthHeader::InvalidFormat(
+                            "Authorization Header should have 2 arguments formatted as \
+                            `Bearer <token>`. Number of arguments did not match the \
+                            number of arguments expected.".to_string())
+                        ),
+                    ))
+                } else {
+                    // Find the FirebaseAuth state
+                    match req.rocket().state::<FirebaseAuth>() {
+                        // Verify if the token is valid
+                        Some(auth) => match auth.verify(header_content).await {
+                            Ok(t) => {
+                                // Token is valid
+                                let firebase_token = FirebaseToken { ..t };
+                                request::Outcome::Success(firebase_token)
+                            }
+                            Err(_) => {
+                                // TODO: Find a way to convert `e` into InvalidJwt error type
+                                // and replace `InvalidJwt::Unspecified` with it
+                                request::Outcome::Failure((
+                                    // Token is invalid
+                                    Status::Unauthorized,
+                                    Error::InvalidJwt(InvalidJwt::Unspecified),
+                                ))
+                            }
+                        },
+                        // FirebaseAuth state not found
+                        None => request::Outcome::Failure((
+                            Status::InternalServerError,
+                            Error::FirebaseAuthStateNotFound,
+                        )),
+                    }
+                }
             }
-            _ => outcome::Outcome::Failure((
-                Status::BadRequest,
-                Error::InvalidAuthHeader(InvalidAuthHeader::BadCount),
+            None => request::Outcome::Failure((
+                Status::Unauthorized,
+                Error::InvalidAuthHeader(InvalidAuthHeader::MissingAuthHeader),
```

Comment:
I think it would be better to distinguish between a 400 bad request error for requests missing `Bearer` tokens or `Authorization` headers. What do you think?

Reply:
I think it's a good idea. I know in ASP.NET it returns 401 when you don't put your cookie or JWT header.